Index: twisted/conch/insults/helper.py
===================================================================
--- twisted/conch/insults/helper.py	(revision 19328)
+++ twisted/conch/insults/helper.py	(working copy)
@@ -199,6 +199,17 @@
             except KeyError:
                 pass
 
+    def setPrivateModes(self, modes):
+        for m in modes:
+            self.privateModes[m] = True
+
+    def resetPrivateModes(self, modes):
+        for m in modes:
+            try:
+                del self.privateModes[m]
+            except KeyError:
+                pass
+
     def applicationKeypadMode(self):
         self.keypadMode = 'app'
 
@@ -306,6 +317,8 @@
         self.home = insults.Vector(0, 0)
         self.x = self.y = 0
         self.modes = {}
+        self.privateModes = {}
+        self.setPrivateModes([insults.privateModes.AUTO_WRAP, insults.privateModes.CURSOR_MODE])
         self.numericKeypad = 'app'
         self.activeCharset = insults.G0
         self.graphicRendition = {
Index: twisted/conch/insults/window.py
===================================================================
--- twisted/conch/insults/window.py	(revision 19328)
+++ twisted/conch/insults/window.py	(working copy)
@@ -43,6 +43,9 @@
         return self.terminal.write(bytes)
 
 class Widget(object):
+    """
+    A visual element.
+    """
     focused = False
     parent = None
     dirty = False
